File: AIQuotaBar/Services/ProviderLocalActivitySnapshot.swift
import Foundation

/// Activity snapshot from a passive local detector (ZCode, MiniMax CLI,
/// Claude Code). Only lifecycle metadata is read; message bodies and tool
/// output are never retained.
///
/// Session IDs are pre-prefixed with their provider and client (for example
/// `glm:zcode:<id>` or `minimax:claude:<id>`) so the protection coordinator
/// can merge, count, and label them without knowing each client's storage
/// format.
struct ProviderLocalActivitySnapshot: Equatable, Sendable {
    var activeSessionIDs: Set<String>
    var lastEventAt: Date?
    var lastEventBySession: [String: Date]

    init(
        activeSessionIDs: Set<String>,
        lastEventAt: Date?,
        lastEventBySession: [String: Date] = [:]
    ) {
        self.activeSessionIDs = activeSessionIDs
        self.lastEventAt = lastEventAt
        self.lastEventBySession = lastEventBySession
    }

    static let empty = ProviderLocalActivitySnapshot(
        activeSessionIDs: [],
        lastEventAt: nil)
}

protocol ProviderLocalActivityProviding: Sendable {
    func snapshot() async -> ProviderLocalActivitySnapshot
}

File: windows/src/AIQuotaBar.Providers/Minimax/ProviderLocalActivitySnapshot.cs
// Swift 来源：AIQuotaBar/Services/ProviderLocalActivitySnapshot.swift — struct ProviderLocalActivitySnapshot。
//
// TODO(W1): Swift 侧该类型是跨 provider 共享（ZCode / Claude Code / MiniMax），归属 Core；
// 本目录不拥有 Core / Providers 根，暂落在 Minimax/ 下。编排者落地 Core 版本时应把本记录上移并让
// 各 provider 检测器引用之（届时删除本文件即可，测试同步改命名空间）。

namespace AIQuotaBar.Providers.Minimax;

/// <summary>
/// 被动本地检测器的活动快照（Swift: ProviderLocalActivitySnapshot）。只读生命周期元数据；
/// 消息体与工具输出永不保留。会话 ID 预先带上 provider 与客户端前缀（如 minimax:cli:&lt;id&gt;）。
/// </summary>
/// <param name="ActiveSessionIds">新鲜度窗口内的会话 ID 集合。</param>
/// <param name="LastEventAt">全部会话中最近一次事件时间。</param>
/// <param name="LastEventBySession">每会话最近事件时间。</param>
public sealed record ProviderLocalActivitySnapshot(
    IReadOnlyCollection<string> ActiveSessionIds,
    DateTimeOffset? LastEventAt,
    IReadOnlyDictionary<string, DateTimeOffset> LastEventBySession)
{
    /// <summary>Swift: ProviderLocalActivitySnapshot.empty。</summary>
    public static ProviderLocalActivitySnapshot Empty { get; } = new(
        Array.Empty<string>(),
        null,
        new Dictionary<string, DateTimeOffset>());
}
